### PR TITLE
[SYCL][ESIMD] Add atomic load/store tests

### PR DESCRIPTION
**Pull request by fineg74**
Complementary compiler PR https://github.com/intel/llvm/pull/7823

### CODE REVIEW

**Review by v-klochkov — APPROVED** [review]

File: SYCL/ESIMD/lsc/atomic_smoke.cpp [hunk 1]
```diff
@@ -576,6 +603,13 @@ int main(void) {
   passed &= test<float, 8, ImplLSCFcmpwr>(q, cfg);
 #endif // USE_DWORD_ATOMICS
 #endif // CMPXCHG_TEST
+
+#ifndef USE_DWORD_ATOMICS
```
Comment (v-klochkov) [inline SYCL/ESIMD/lsc/atomic_smoke.cpp:607]: Can you explain please, why these new tests are limited to LSC atomic ops only?
As I can see both enums sycl::ext::intel::esimd::native::lsc::atomic_op and sycl::ext::intel::esimd::atomic_op have 'load' and 'store' operations.

**Comment by fineg74** [comment]
Test failure for windows:
SYCL :: SubGroupMask/Basic.cpp
Is not related to the change

File: SYCL/ESIMD/lsc/atomic_smoke.cpp (hunk 1 above)
Comment (fineg74) [reply SYCL/ESIMD/lsc/atomic_smoke.cpp:607]: We don't have an intrinsic to correctly handle non LSC version of atomic_update for load/store leading to crash of sycl_post-link

Comment (v-klochkov) [reply SYCL/ESIMD/lsc/atomic_smoke.cpp:607]: Ok, thank you!

**Comment by v-klochkov** [comment]
> Test failure for windows: SYCL :: SubGroupMask/Basic.cpp Is not related to the change

Per this message it is known and tracked: https://github.com/intel/llvm-test-suite/pull/1487#issuecomment-1372380048

**Pull request merged**